mesonbuild/environment.py

```diff
@@ -572,6 +573,106 @@ def __init__(self, source_dir: T.Optional[str], build_dir: T.Optional[str], opti
         self.default_cmake = ['cmake']
         self.default_pkgconfig = ['pkg-config']
         self.wrap_resolver = None
+        self.pkgconfig_repo = None
+
+    @staticmethod
+    def create_pkgconfig_repo(for_machine: MachineChoice = MachineChoice.BUILD,
+                              is_cross: bool = False,
+                              system_paths: T.Optional[T.List[str]] = None,
+                              extra_paths: T.Optional[T.List[str]] = None,
+                              sysroot_dir: T.Optional[str] = None)  -> pkgconfig.Repository:
+        # Get some default paths. This is normally defined by pkg-config packager,
+        # it could need per distro/platform tweaking in the future.
+        default_prefixes = [Path('/usr/local'), Path('/usr')]
```

Comment:
Would it make sense to look at ~/.local as well?

Reply:
I don't think any distro/platform configure pkg-config/pkgconf to include that path, is there? I'm not against it if there is use-case for it? In any case, it can already be done by setting PKG_CONFIG_PATH.

Reply:
If you are only going for distro config, then this is good as is. ~/.local would be for extra convenience, but mimicking distros sounds like a good idea to me.